sensibo/unified.js

```diff
@@ -1,235 +1,485 @@
 				}
 			}
 
 			// set fanSpeeds
 			if (modeCapabilities.fanLevels && modeCapabilities.fanLevels.length) {
-				capabilities[mode].fanSpeeds = modeCapabilities.fanLevels
+				mode.fanSpeeds = modeCapabilities.fanLevels
 
 				// set AUTO fanSpeed
-				if (capabilities[mode].fanSpeeds.includes('auto')) {
-					capabilities[mode].autoFanSpeed = true
+				if (mode.fanSpeeds.includes('auto')) {
+					mode.autoFanSpeed = true
 				} else {
-					capabilities[mode].autoFanSpeed = false
+					mode.autoFanSpeed = false
 				}
 			}
 
 			// set vertical swing
 			if (modeCapabilities.swing) {
 				if (modeCapabilities.swing.includes('both')) {
-					capabilities[mode].horizontalSwing = true
-					capabilities[mode].verticalSwing = true
-					capabilities[mode].threeDimensionalSwing = true
+					mode.horizontalSwing = true
+					mode.verticalSwing = true
+					mode.threeDimensionalSwing = true
 				} else {
 					if (modeCapabilities.swing.includes('rangeFull')) {
-						capabilities[mode].verticalSwing = true
+						mode.verticalSwing = true
+						mode.threeDimensionalSwing = false
 					}
 
 					if (modeCapabilities.swing.includes('horizontal')) {
-						capabilities[mode].horizontalSwing = true
+						mode.horizontalSwing = true
+						mode.threeDimensionalSwing = false
 					}
 				}
 			}
 
 			// set horizontal swing
-			if (!capabilities[mode].horizontalSwing && modeCapabilities.horizontalSwing && modeCapabilities.horizontalSwing.includes('rangeFull')) {
-				capabilities[mode].horizontalSwing = true
+			if (!capabilities[modeString].horizontalSwing &&
+				modeCapabilities.horizontalSwing &&
+				modeCapabilities.horizontalSwing.includes('rangeFull')) {
+				mode.horizontalSwing = true
 			}
 
 			// set light
 			if (modeCapabilities.light) {
-				capabilities[mode].light = true
+				mode.light = true
 			}
 
-			platform.log.easyDebug(`${roomName} - Mode: ${mode}, Capabilities: `)
-			platform.log.easyDebug(capabilities[mode])
+			platform.easyDebugInfo(`${device.room.name} - Mode: ${modeString}, Capabilities:`)
+			platform.easyDebugInfo(JSON.stringify(mode, null, 4))
 		}
 
 		return capabilities
 	},
 
-	acState: device => {
-		const state = {
-			active: device.acState.on,
-			mode: device.acState.mode.toUpperCase(),
-			targetTemperature: !device.acState.targetTemperature ? null : device.acState.temperatureUnit === 'C' ? device.acState.targetTemperature : toCelsius(device.acState.targetTemperature),
-			currentTemperature: device.measurements.temperature,
-			relativeHumidity: device.measurements.humidity,
-			smartMode: device.smartMode,
-			light: device.acState.light && device.acState.light !== 'off',
-			pureBoost: device.pureBoostConfig && device.pureBoostConfig.enabled
+	/**
+	 * @param {import('../types').Device} device
+	 * @returns {import('../types').InternalSmartModeState}
+	 */
+	getInternalSmartModeState: function (device)  {
+		/** @type {null|import('../types').InternalSmartModeState} */
+		let smartModeState = null
```

Comment:
Are you explicitly expecting getInternalSmartModeState to be null in some cases?

Otherwise I'm thinking maybe this should be:
```suggestion
		const smartModeState = {}
```